sys/aarch64/boot.c

```diff
@@ -199,6 +208,39 @@ __boot_text static paddr_t build_page_table(void) {
 
   l0[L0_INDEX(DMAP_BASE)] = (pde_t)l1d | L0_TABLE;
 
+#if KASAN /* Prepare KASAN shadow mappings */
+  size_t kasan_sanitzied_size =
+    SUPERPAGESIZE + roundup2(va - KASAN_MD_SANITIZED_START,
+                             SUPERPAGESIZE * KASAN_SHADOW_SCALE_SIZE);
+  size_t kasan_shadow_size = kasan_sanitzied_size / KASAN_SHADOW_SCALE_SIZE;
+  vaddr_t kasan_shadow_end = KASAN_MD_SHADOW_START + kasan_shadow_size;
+  va = KASAN_MD_SHADOW_START;
+  kasan_sanitized_end = KASAN_MD_SANITIZED_START + kasan_sanitzied_size;
+  /* Allocate physical memory for shadow area */
+  pa = (paddr_t)bootmem_alloc(kasan_shadow_size);
+
+  while (va < kasan_shadow_end) {
```

Comment:
Maybe you should add more comments to this loop and checks inside. At least what whole loop is doing.
Variable name `va` hardly suggests that it's about "start" of some kasan_shadow, you should rename it.

Reply:
No I shouldn't - it's standard walk through page table and function name is `build_page_table`. There is noting more to say about that.
`va` is standard name for virtual address and it __is__ a virtual address. It's common pattern to use `va` in boot and pmap code.